File: src/main/java/social/alone/server/event/service/EventDeleteService.java
package social.alone.server.event.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import social.alone.server.event.Event;
import social.alone.server.event.repository.EventRepository;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class EventDeleteService {

    private final EventRepository eventRepository;

    public void delete(Long eventId) {
        Optional<Event> event = this.eventRepository.findById(eventId);
        event.ifPresent(this.eventRepository::delete);
    }
}

File: src/main/kotlin/social/alone/server/event/service/EventDeleteService.kt
package social.alone.server.event.service

import org.springframework.stereotype.Service
import social.alone.server.event.domain.Event
import social.alone.server.event.repository.EventRepository
import java.util.function.Consumer

@Service
class EventDeleteService(private val eventRepository: EventRepository) {

    fun delete(eventId: Long?) {
        val event = this.eventRepository!!.findById(eventId!!)
        event.ifPresent(Consumer<Event> { this.eventRepository.delete(it) })
    }
}
